lib/Dialect/OM/OMOps.cpp

```diff
@@ -280,6 +280,38 @@ OpFoldResult circt::om::ConstantOp::fold(FoldAdaptor adaptor) {
   return getValueAttr();
 }
 
+//===----------------------------------------------------------------------===//
+// ListCreateOp
+//===----------------------------------------------------------------------===//
+
+void circt::om::ListCreateOp::print(OpAsmPrinter &p) {
+  p << " ";
+  p.printOperands(getInputs());
+  p.printOptionalAttrDict((*this)->getAttrs());
+  p << " : " << getInputs()[0].getType();
+}
+
+ParseResult circt::om::ListCreateOp::parse(OpAsmParser &parser,
+                                           OperationState &state) {
+  llvm::SMLoc loc = parser.getCurrentLocation();
+  llvm::SmallVector<OpAsmParser::UnresolvedOperand, 16> operands;
+  Type elemType;
+
+  if (parser.parseOperandList(operands) ||
+      parser.parseOptionalAttrDict(state.attributes) || parser.parseColon() ||
+      parser.parseType(elemType))
+    return failure();
+
+  if (operands.size() == 0)
+    return parser.emitError(loc, "Cannot construct a list of length 0");
```

Comment:
I'm wondering if it might be more useful to allow empty list construction by this op. WDYT?